renku/core/incubation/command.py

```diff
@@ -123,20 +123,23 @@ def execute(self, *args, **kwargs):
 
         output = None
         error = None
+
         try:
             with context["stack"]:
                 output = context["click_context"].invoke(self._operation, context["client"], *args, **kwargs)
         except errors.RenkuException as e:
             error = e
+            if raise_exceptions:
```

Comment:
The original idea was that the post hooks can handle exceptions if they want (and even remove them from the result if they don't matter). This is important for e.g. being able to reset the working directory to the previous state on error to not leave it dirty and things like that.

But I see the need to raise exceptions in general, especially unhandled ones. It probably makes sense to have a post-hook that raises any unhandled exception and runs as the last hook of the whole chain. 

What do you think?

Reply:
That's a great idea! I added a _post_hook to do this. However, I've changed the order that post-hooks execute to be descending, so when command's pre-hook runs first it's post-hook runs last. Let me know if you think it should not be done in this way.